### DISCUSSION TITLE
Custom fonts with transparent back ground

### DISCUSSION THREAD

**Discussion by karlheinz2000** [Bodmer/TFT_eSPI#1084] [Q&A - Fonts] [answered]
I have generated custom fonts and can `drawString `as long as text color is defined as `setTextColor(uint16_t fgcolor, uint16_t bgcolor)`. If I only specify `setTextColor(uint16_t color)`, to get transparent bg, font seems to be replaced by some default font (with transparent bg).

**Comment by Bodmer** [comment]
I changed the example Free_Font_Demo with no background and it works fine.
Are you using a smooth font?

Provide a complete sketch that demonstrates the probem.
State which version of the library you are using.

**Comment by karlheinz2000** [comment]
Edit: same with 2.3.61

Lib version is 2.3.59
I created a font as mentioned.

This is test sketch:

```
#include "SPI.h"
#include "TFT_eSPI.h"

#include "Calibri_32.h"

// Use hardware SPI
TFT_eSPI tft = TFT_eSPI();

void setup(void) {

  tft.begin();
  tft.setRotation(3);

}

void loop() {

  tft.fillScreen(TFT_BLACK);

  tft.loadFont(Calibri_32);
  tft.setTextDatum(TL_DATUM);
 
  tft.setTextColor(TFT_WHITE);
  tft.drawString("setTextColor(TFT_WHITE)", 0, 50);

  tft.setTextColor(TFT_WHITE, TFT_BLACK);
  tft.drawString("setTextColor(TFT_WHITE, TFT_BLACK)", 0, 150);

  while (1);
}
```

Ok, seems not like a different font, but result is different. Not easy to see on photo, but check for s,e,o.
For smaller fonts it was looking like a scaled default font.
![DSC_0023](https://user-images.githubusercontent.com/11009766/111864974-6e962280-8964-11eb-83f0-b8b2f8328efa.JPG)

**Comment by Bodmer** [comment, marked as answer]
You have got to specify the background colour for antialiased fonts as the edges are blended with the background. If you do not specify the background then the pixels that should be blended show as foreground, so the anti-aliasing in lost and the characters appear to change shape.

I will be adding a new attribute switch to turn on/off the background whilst still maintaining anti-aliasing. This is partially [implemented in the beta branch](https://github.com/Bodmer/TFT_eSPI/blob/2.4.0-Beta/TFT_eSPI.h#L679-L683). Commands will be:
```
tft.setAttribute(SFBG_ENABLE, false); // Turn off smooth font background rendering
tft.setAttribute(SFBG_ENABLE, true);  // Turn on smooth font background rendering
```
However there is a bug in this new feature so it has not been incorporated into the master yet.

**Reply by karlheinz2000** [reply]
Thanks for quick reply and the solution provided.
In the meantime I used sprites to fix this.
